tests/tests_basics_files2.yml

```diff
@@ -53,42 +74,33 @@
 
     - name: Check file counts in rsyslog.d
       assert:
-        that: rsyslog_d_file_count.matched >= 6
+        that: rsyslog_d_file_count.matched == 8
 
     # Checking 'error' in stdout from systemctl status is for detecting the case in which rsyslog is running,
     # but some functionality is disabled due to some error, e.g., error: 'tls.cacert' file couldn't be accessed.
     - name: Check rsyslog errors
       command: systemctl status rsyslog
-      register: systemctl_result
-      failed_when: "'error' in systemctl_result.stdout or 'a RainerScript command' in systemctl_result.stdout or systemctl_result.rc != 0"
+      register: __result
+      failed_when: "'error' in __result.stdout or 'a RainerScript command' in __result.stdout or __result.rc != 0"
 
-    - name: Get the files config stat
+    - name: To verify the generated filename is correct, check the config file of files output exists
       stat:
         path: "{{ __test_files_conf }}"
-      register: stat_output_files_system
+      register: __result
+      failed_when: not __result.stat.exists
 
-    - name: Check if the files config exists
-      assert:
-        that: stat_output_files_system.stat.exists
-
-    - name: Grep output to messages line
-      command: /bin/grep ".var.log.messages" {{ __test_files_conf }}
-      register: files_conf_messages
+    - name: Check the config file contains the expected filter and the output file as configured.
+      command: /bin/grep '\*.info;authpriv.none;auth.none;cron.none;mail.none.*{{ __default_system_log }}' '{{ __test_files_conf }}'
+      register: __result
       changed_when: false
-
-    - name: Check the filter
-      assert:
-        that: files_conf_messages.stdout is match(" *\*.info;*authpriv.none;auth.none;cron.none;mail.none.*.var.log.messages")
+      failed_when: __result.rc != 0
```

Comment:
Use `failed_when: __result is failed` - see https://docs.ansible.com/ansible/latest/user_guide/playbooks_tests.html#task-results

But for this case, do you need a `register` and a `failed_when`?  That is, if `command` returns with an error, the task will fail, and you don't use `__result` elsewhere.